Building micro_speech for Arm Mbed OS using ARM Compiler (armclang) fails

@tensorflow/micro

**System information**
- Host OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- TensorFlow installed from (source or binary): source
- Tensorflow version (commit SHA if source): 917882bfa2
- Target platform (e.g. Arm Mbed OS, Arduino Nano 33 etc.): Arm Mbed OS

**Describe the problem**
Building micro_speech with the commands below results in this error:
`./tensorflow/lite/micro/tools/make/downloads/kissfft/kiss_fft.h:45:10: fatal error: 'sys/types.h' file not found`

Switching out ARMC6 for GCC_ARM in the final command fixes the issue.
The issue comes from sys/types.h not being available in the ARM Compiler toolchain, while it is available when using gcc. 

Patching kissfft in the download_and_extract.sh-script to remove the `#include <sys/types.h>` fixes the issue (https://github.com/jenselofsson/tensorflow/commit/080bc19079f25a2f2151e67d303f1f1949bcfc3e), and it compiles using both GCC_ARM and ARMC6.

**Please provide the exact sequence of commands/steps when you ran into the problem**
$ make -f tensorflow/lite/micro/tools/make/Makefile TAGS="cmsis-nn disco_f746ng" generate_micro_speech_mbed_project
$ cd tensorflow/lite/micro/tools/make/gen/linux_x86_64/prj/micro_speech/mbed/
$ mbed config root .
$ mbed deploy
$ mbed compile -m DISCO_F746NG -t ARMC6

--- a/tensorflow/lite/micro/tools/make/download_and_extract.sh
+++ b/tensorflow/lite/micro/tools/make/download_and_extract.sh
@@ -72,6 +72,7 @@ patch_am_sdk() {
 patch_kissfft() {
   sed -i -E $'s@#ifdef FIXED_POINT@// Patched automatically by download_dependencies.sh so default is 16 bit.\\\n#ifndef FIXED_POINT\\\n#define FIXED_POINT (16)\\\n#endif\\\n// End patch.\\\n\\\n#ifdef FIXED_POINT@g' tensorflow/lite/micro/tools/make/downloads/kissfft/kiss_fft.h
 
+  sed -i -E '/^#include <sys\/types.h>/d' tensorflow/lite/micro/tools/make/downloads/kissfft/kiss_fft.h
   # Fix for https://github.com/mborgerding/kissfft/issues/20
   sed -i -E $'s@#ifdef FIXED_POINT@#ifdef FIXED_POINT\\\n#include <stdint.h> /* Patched. */@g' tensorflow/lite/micro/tools/make/downloads/kissfft/kiss_fft.h
 

